types/schema.d.ts

```diff
@@ -20,7 +20,16 @@ export interface FastifyRouteSchemaDef {
   httpStatus?: string;
 }
 
+export interface FastifyValidationResult {
+  errors?: FastifySchemaValidationError[];
+}
+
+export interface FastifySchemaValidationError {
+  message: string;
+  dataPath: string;
+}
+
 /**
  * Compiler for FastifySchema Type
  */
-export type FastifySchemaCompiler = (routeSchema: FastifyRouteSchemaDef) => unknown
+export type FastifySchemaCompiler = (routeSchema: FastifyRouteSchemaDef) => FastifyValidationResult
```

Comment:
Is this a bug? It looks like this should be like this instead:

```ts
export type FastifySchemaCompiler = (routeSchema: FastifyRouteSchemaDef) => (data: unknown) => FastifyValidationResult
```

As it is now, the signature suggest the validation result is directly derived from the schema by the compiler but actually the compiler returns a function that takes the data as input to generate the validation result instead.